build/parsePrep.c

```diff
@@ -97,9 +97,10 @@ static char *doPatch(rpmSpec spec, uint32_t c, int strip, const char *db,
     /* Avoid the extra cost of fork and pipe for uncompressed patches */
     if (compressed != COMPRESSED_NOT) {
 	patchcmd = rpmExpand("{ %{uncompress: ", fn, "} || echo patch_fail ; } | "
-                             "%{__patch} ", args, NULL);
+                         "RPM_PATCH_NAME=", sp->path,
+                             " %{__patch} ", args, NULL);
     } else {
-	patchcmd = rpmExpand("%{__patch} ", args, " < ", fn, NULL);
+	patchcmd = rpmExpand("RPM_PATCH_NAME=", sp->path, " %{__patch} ", args, " < ", fn, NULL);
```

Comment:
`sp->path` is the same as `fn`, and it's what's passed to `stdin` of the `%__patch` binary, which means you can easily get that value by running `readlink -f /dev/stdin` in your binary. Handy, isn't it? :)